server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -871,6 +836,53 @@ private void cleanupFiles(
         }
     }
 
+    /**
+     * Segment replication method
+     *
+     * This method takes the segment info bytes to build SegmentInfos. It inc'refs files pointed by passed in SegmentInfos
+     * bytes to ensure they are not deleted.
+     *
+     * @param tmpToFileName Map of temporary replication file to actual file name
+     * @param infosBytes bytes[] of SegmentInfos supposed to be sent over by primary excluding segment_N file
+     * @param segmentsGen segment generation number
+     * @param consumer consumer for generated SegmentInfos
+     * @throws IOException Exception while reading store and building segment infos
+     */
+    public void buildInfosFromStore(
```

Comment:
nit - I think we should rename this to indicate the infos is provided via replication vs from store, since it is not yet part of a commit on store.  `buildInfosFromBytes` ?

As a follow up it would be great to reuse this logic with remote store with https://github.com/opensearch-project/OpenSearch/issues/7802

Reply:
Updated.